File: Android-Core/01-AndroidCore/e-Intent/01_IntentBasic/02_transferobject/src/main/java/top/iqqcode/transferobject/Person.java
package top.iqqcode.transferobject;

import java.io.Serializable;

/**
 * @Author: iqqcode
 * @Date: 2021-03-07 10:12
 * @Description:
 */
public class Person implements Serializable {
    private String name;
    private int age;

    public Person() {}

    public Person(String name, int age) {
        this.name = name;
        this.age = age;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public int getAge() {
        return age;
    }

    public void setAge(int age) {
        this.age = age;
    }
}

File: Kotlin/KotlinKernelGo/src/main/kotlin/top/iqqcode/collection/Person.kt
package top.iqqcode.collection

/**
 * @Author: iqqcode
 * @Date: 2023-09-19 02:01
 * @Description:
 */
open class Person(val name: String, var age: Int) {
    val isAdult: Boolean
        get() { // do something else
            return age >= 18
        }

    open val canWalk: Boolean = false
    open fun walk() {}
}

class Boy : Person("", 1) {

    override val canWalk: Boolean = true
    override fun walk() {}

}